frontend/src/components/layout/Sidebar.tsx

```diff
@@ -0,0 +1,179 @@
+'use client';
+
+import React, { useState } from 'react';
+import { useUser } from "@/components/UserContext";
+import { ChevronDown, ChevronRight, Home, File, FolderClosed, Plus, PanelLeftClose, PanelLeft } from 'lucide-react';
+import { useTranslation } from 'react-i18next';
+import UploadDialog from '@/components/upload/UploadDialog';
+import Loading from "@/app/[locale]/loading";
+import ItemList from '@/components/layout/ItemList';
+import type { Course, Document } from "@/types/entities";
+import {useFavorites} from "@/hooks/useFavorites";
+
+const mapCoursesToItems = (courses: Course[]) => {
+  return courses.map(course => ({
+    name: course.name,
+    code: course.code,
+    redirectUrl: `/courses/${course.id}<itemList
+                title={t('account.favorite.courses')}
+                items={mapCoursesToItems(user!.favoriteCourses!)}
+                emptyMessage={t('account.favorite.no_courses')}
+      />`
+  }));
+};
+
+const mapDocumentsToItems = (documents: Document[]) => {
+  return documents.map(document => ({
+    name: document.name,
+    redirectUrl: `/documents/${document.id}`
+  }));
+};
+
+function handleLogout() {
+  // TODO implement logout
+}
+
+const NavigationSidebar = () => {
+  const { user, loading } = useUser();
+  const { t, i18n } = useTranslation();
+  const [isCollapsed, setIsCollapsed] = useState(false);
+  const [expandedSections, setExpandedSections] = useState({
+    courses: true,
+    documents: false
+  });
+  const [isUploadDialogOpen, setIsUploadDialogOpen] = useState(false);
+  const {
+    updateFavoriteCourse,
+    updateFavoriteDocument
+  } = useFavorites(user);
+
+  const toggleSection = (section: keyof typeof expandedSections) => {
+    setExpandedSections(prev => ({
+      ...prev,
+      [section]: !prev[section]
+    }));
+  };
+
+  const handleUploadButtonClick = () => {
+    setIsUploadDialogOpen(true);
+  };
+
+  const handleUploadDialogClose = () => {
+    setIsUploadDialogOpen(false);
+  };
+
+  if (loading) {
+    return <Loading />;
+  }
+
+  return (
+      <div className={`relative transition-all duration-300 ${isCollapsed ? 'w-16' : 'w-64'} h-screen bg-white border-r border-gray-200 flex flex-col`}>
+        {/* Collapse Toggle Button */}
+        <button
+            onClick={() => setIsCollapsed(!isCollapsed)}
+            className="absolute -right-3 top-4 bg-white border border-gray-200 rounded-full p-1 hover:bg-gray-100 "
+            aria-label={'toggle'}
+        >
+          {isCollapsed ? <PanelLeft size={16} /> : <PanelLeftClose size={16} />}
+        </button>
+
+        {/* Top Navigation */}
+        <nav className="p-4 space-y-2">
+          <a href={`/${i18n.language}`} className="flex items-center space-x-2 text-gray-700 hover:bg-gray-100 rounded p-2 transition-all duration-200 hover:scale-[1.01]">
+            <Home size={20} />
+            {!isCollapsed && <span>{t('sidebar.home')}</span>}
+          </a>
```

Comment:
Use Link instead of a
```suggestion
          <Link href={`/${i18n.language}`} className="flex items-center space-x-2 text-gray-700 hover:bg-gray-100 rounded p-2 transition-all duration-200 hover:scale-[1.01]">
            <Home size={20} />
            {!isCollapsed && <span>{t('sidebar.home')}</span>}
          </Link>
```